### PR TITLE
Add support for the connection status API

### PR DESCRIPTION
**Pull request by Hawxy**
### Changes

Adds the connection status check API as I noticed it was missing from the connections client.

### References

https://auth0.com/docs/api/management/v2#!/Connections/get_status

### Testing

Not sure of a good way of adding tests for this, as a valid AD or LDAP connection would need to be setup on the Auth0 side.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All existing and new tests complete without errors

### CODE REVIEW

File: src/Auth0.ManagementApi/Clients/ConnectionsClient.cs [hunk 1]
```diff
@@ -132,5 +132,16 @@ public Task<Connection> UpdateAsync(string id, ConnectionUpdateRequest request,
         {
             return Connection.SendAsync<Connection>(new HttpMethod("PATCH"), BuildUri($"connections/{EncodePath(id)}"), request, DefaultHeaders, cancellationToken: cancellationToken);
         }
+
+        /// <summary>
+        /// Retrieves the status of an ad/ldap connection 
+        /// </summary>
+        /// <param name="id">D of the connection to check</param>
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Clients/ConnectionsClient.cs:139]:
```suggestion
        /// <param name="id">ID of the connection to check</param>
```

File: src/Auth0.ManagementApi/Clients/ConnectionsClient.cs [hunk 2]
```diff
@@ -132,5 +132,16 @@ public Task<Connection> UpdateAsync(string id, ConnectionUpdateRequest request,
...
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous check operation. Will throw if the check fails.</returns>
+        public Task CheckAsync(string id, CancellationToken cancellationToken = default)
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Clients/ConnectionsClient.cs:142]:
```suggestion
        public Task CheckStatusAsync(string id, CancellationToken cancellationToken = default)
```

File: src/Auth0.ManagementApi/Clients/IConnectionsClient.cs [hunk 3]
```diff
@@ -62,5 +62,13 @@ public interface IConnectionsClient
...
     /// <returns>The <see cref="Connection"/> that has been updated.</returns>
     Task<Connection> UpdateAsync(string id, ConnectionUpdateRequest request, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Retrieves the status of an ad/ldap connection.
+    /// </summary>
+    /// <param name="id">ID of the connection to check.</param>
+    /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
+    /// <returns>A <see cref="Task"/> that represents the asynchronous check operation. Will throw if the status check fails.</returns>
+    Task CheckAsync(string id, CancellationToken cancellationToken = default);
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Clients/IConnectionsClient.cs:72]:
```suggestion
    Task CheckStatusAsync(string id, CancellationToken cancellationToken = default);
```

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by Hawxy** [comment]
PR updated with code review items.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**